In [1]:
import pandas as pd
import numpy as np 
import warnings
warnings.filterwarnings('ignore')

In [2]:
df=pd.read_excel("D:\sahil\project data trained\India_Crimeforproject.xlsx")
df

STATE/UT  YEAR    Region  Literacy_Rate  Population   \
0    A & N ISLANDS  2001  Southern          81.30     356152.0   
1    A & N ISLANDS  2002  Southern          81.80     358531.0   
2    A & N ISLANDS  2003  Southern          82.30     360910.0   
3    A & N ISLANDS  2004  Southern          82.80     363290.0   
4    A & N ISLANDS  2005  Southern          83.30     365669.0   
..             ...   ...       ...            ...          ...   
415    WEST BENGAL  2008   Eastern          80.09   87239781.0   
416    WEST BENGAL  2009   Eastern          80.62   88248865.0   
417    WEST BENGAL  2010   Eastern          81.16   89257948.0   
418    WEST BENGAL  2011   Eastern          81.69   90267032.0   
419    WEST BENGAL  2012   Eastern          82.87   91276115.0   

     total literate population  Area (km²)  MURDER  ATTEMPT TO MURDER  \
0                       289552        8249      13                  0   
1                       293278        8249      14                  3   
2                       297029        8249      16                  4   
3                       300804        8249      13                  1   
4                       304602        8249      14                  3   
..                         ...         ...     ...                ...   
415                   69870341       88752     192                168   
416                   71146235       88752     229                238   
417                   72441751       88752     244                224   
418                   73739138       88752     232                215   
419                   75640517       88752     140                108   

     CULPABLE HOMICIDE NOT AMOUNTING TO MURDER  ...  \
0                                            0  ...   
1                                            0  ...   
2                                            1  ...   
3                                            2  ...   
4                                            3  ...   
..                                         ...  ...   
415                                         47  ...   
416                                        241  ...   
417                                        179  ...   
418                                        126  ...   
419                                         80  ...   

     INSULT TO MODESTY OF WOMEN  CRUELTY BY HUSBAND OR HIS RELATIVES  \
0                             1                                    9   
1                             3                                    4   
2                             2                                    7   
3                             3                                    5   
4                             1                                    5   
..                          ...                                  ...   
415                           5                                 1946   
416                           3                                 2225   
417                           8                                 2378   
418                          10                                 2847   
419                           1                                 1860   

     IMPORTATION OF GIRLS FROM FOREIGN COUNTRIES  CAUSING DEATH BY NEGLIGENCE  \
0                                              0                            0   
1                                              0                            0   
2                                              0                            0   
3                                              0                            6   
4                                              0                            3   
..                                           ...                          ...   
415                                            0                           12   
416                                            0                            4   
417                                            5                           11   
418 

The population, geographic area, and literacy rate of each state have all been carefully compiled from various websites. There was data available for each state's population, area, and literacy rate for the years 2001 and 2011. As a result, the necessary computations and assumptions were done to obtain the data for the years 2002 through 2010 and 2012. As the population of Andhra Pradesh declined by approximately 11% between 2001 and 2011, we needed to expand by approximately 1% year to attain the same result. A number of websites were consulted in order to retrieve the same data. Among them are:

https://en.wikipedia.org/wiki/List_of_states_in_India_by_past_population
https://en.wikipedia.org/wiki/List_of_states_and_union_territories_of_India_by_area
https://en.wikipedia.org/wiki/List_of_Indian_states_and_union_territories_by_literacy_rate
https://socialjustice.gov.in/common/76669
http://cyberjournalist.org.in/census/cenindia.html
https://www.indiastat.com/specimen-tables/geographical-data
https://www.wbhealth.gov.in/other_files/2007/14_5.html
https://en.wikipedia.org/wiki/List_of_states_and_union_territories_of_India_by_population